src/main/java/com/tenx/ms/retail/product/repository/ProductRepository.java

```diff
@@ -0,0 +1,17 @@
+package com.tenx.ms.retail.product.repository;
+
+
+import com.tenx.ms.retail.product.domain.ProductEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface ProductRepository extends JpaRepository<ProductEntity, Long> {
+
+    List<ProductEntity> findAllByStoreStoreId(Long storeId);
```

Comment:
@marielbysoares I had paging and sorting almost "for free" by using PagingAndSortingRepository instead of JpaRepository. worth seeing.